public/app/features/explore/QueryLibrary/QueryTemplatesList.tsx

```diff
@@ -40,6 +43,25 @@ export function QueryTemplatesList(props: QueryTemplatesListProps) {
         return;
       }
 
+      let userDataList;
+      try {
+        const userQtList = uniq(data.map((qt) => qt.user?.uid));
+        const usersParam = userQtList.map((user) => `key=${user}`).join('&');
```

Comment:
We should probably pass this through https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/encodeURIComponent or is the username guaranteed to be URL safe?

Reply:
this is user uid and since uid is used as the name of the k8s object that doesn't allow special characters it should always be URL safe. However, I've updated this to use `encodeURIComponent` just in case